abstract_specification.md

```diff
@@ -233,68 +298,298 @@ Any given storage "engine" upon which SOMA is implemented may have additional fe
+Objects accessed via getting a Collection element, or objects created with one of the <code>add_new\_<var>object_type</var></code> methods are considered "owned" by the collection.
+All such objects will be automatically closed together by [the collection's close operation](#operation-close-collection-types).
+
+### Operation: create() (Collection types)
+
+Create a new collection of the given type with the user-specified URI.
+The collection type created should match the collection type it is called upon; for example `SOMAExperiment.create` should create a `SOMAExperiment`.
+This `create` function has no parameters beyond [those shared by all `create` methods](#operation-create).
 
-### Operation: create()
+```
+SOMACollectionType.create(string uri, platform_config, context) -> SOMACollectionType
+```
+
+### Operation: close (Collection types)
 
-Create a new SOMACollection with the user-specified URI.
+In addition to the details of the [common close operation](#operation-close), closing has additional semantics specific to collection types.
+
+When a SOMA collection is directly used to instantiate child SOMA objects, those SOMA objects are considered "owned" by the collection, and the collection manages their lifecycle.
+When the collection is closed, it must close any of these child objects before closing itself.
+For example:
 
 ```
-create(string uri, platform_config) -> void
+root = soma_impl.open('path://netloc/root/collection', 'w')
+
+child_a = root['child_a']
+# child_a was reified by root and is thus is owned by root.
+grandchild = child_a['grandchild']
+# grandchild is owned by child_a and in turn by root.
+
+new_child = root.add_new_dataframe('new_child', ...)
+# new_child was created by root and thus is also owned by root.
+
+external_object = soma_impl.open('path://other/data')
+# external_object was created independently and thus has no owner.
+root['other_key'] = external_object
+# Even though external_object has been set as an element of root,
+# the instance `external_object` was not itself created by root
+# and is thus still not owned by root.
+
+root.close()
+# Root will close its members, which will in turn recursively close theirs.
+# The exact order may vary but will still reflect the ownership tree:
+#  1. root is asked to close.
+#      2. root asks child_a to close.
+#          3. child_a asks grandchild to close.
+#              4. grandchild closes itself.
+#          5. child_a closes itself.
+#      6. root asks new_child to close.
+#          7. new_child closes itself.
+#      8. root closes itself.
+#
+# external_object remains open because it is not owned by root.
+```
+
+A user may close a child object before closing the parent object.
+Because `close` is idempotent, this is explicitly allowed.
+Closing the child will _not_ otherwise affect the parent object; it will remain open as usual.
+
+### Operation: set
+
+Adds an entry to the collection, overwriting an existing key if one exists already.
```

Comment:
Does the spec have an opinion on what happens to data originally stored by the object that is replaced at a specified key? Is it orphaned or removed?